code/modules/projectiles/projectile/special.dm

```diff
@@ -387,3 +387,206 @@
 		var/obj/effect/ebeam/B = b
 		animate(B, alpha = 0, time = 32)
 	return ..()
+
+/obj/item/projectile/hallucination
```

Comment:
fireball:
place explosion sound
fake blindness overlay
knock the mob down

would seem p.real for a couple of seconds (aka all hallucinations)